app/uk/gov/hmrc/economiccrimelevyregistration/navigation/DoYouHaveCtUtrPageNavigator.scala

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.economiccrimelevyregistration.navigation
+
+import play.api.mvc.Call
+import uk.gov.hmrc.economiccrimelevyregistration.models.{CheckMode, Mode, NormalMode, Registration}
+
+import javax.inject.Inject
+import scala.concurrent.ExecutionContext
+import uk.gov.hmrc.economiccrimelevyregistration.controllers.routes
+
+class DoYouHaveCtUtrPageNavigator @Inject() (implicit
+  ex: ExecutionContext
+) extends PageNavigator {
+  override protected def navigateInNormalMode(registration: Registration): Call =
+    registration.otherEntityJourneyData.isCtUtrPresent match {
+      case Some(isCtUtrPresent) => navigateInEitherMode(isCtUtrPresent, NormalMode)
+      case None                 => routes.NotableErrorController.answersAreInvalid()
+    }
+
+  override protected def navigateInCheckMode(registration: Registration): Call =
+    registration.otherEntityJourneyData.isCtUtrPresent match {
+      case Some(isCtUtrPresent) => navigateInEitherMode(isCtUtrPresent, CheckMode)
```

Comment:
If in check mode, you always go to check your answers.

Reply:
In this case it's a bit different. If user says "Yes" it should go to `CtUtrController`. If we went straight to check your answers we would never offer him a chance to enter the CT UTR if he said "Yes". Would you agree with that ?